jupiterone/resource_rule.go

```diff
@@ -388,6 +388,20 @@ func (r *QuestionRuleResource) Update(ctx context.Context, req resource.UpdateRe
 	// Read Terraform plan data into the model
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 
+	// The UpdateRule operation needs the most current version of the rule to update it.
+	// We fetch it from the state if it is not specified by the user.
+	if data.Version.ValueInt64() == 0 {
+		paths, _ := req.State.PathMatches(ctx, path.MatchRoot("version"))
+		// I'm not sure what method to use to add this Diagnostic
+		// resp.Diagnostics.Append(d)
+		if len(paths) != 1 {
+			resp.Diagnostics.AddError("failed to get version from state", "no version path found in state")
+		}
+		versionPath := paths[0]
+		// TODO capture diagnostic
+		req.State.GetAttribute(ctx, versionPath, &data.Version)
+	}
```

Comment:
I'll admit, I was having some trouble adding the diagnostics from `PathMatches` and `GetAtrribute` to `resp.Diagnostics`. I'm not sure if this was the correct way to do this, so very open to suggestions.

Reply:
The `version` has be unknown in the plan so that an error isn't reported when a new value is set after the Update, so the `ModifyPlan` method has to set the `version` to unknown explicitly so that the framework knows it is supposed to be updated by the operation.

The 0 is the clue, because that value shouldn't ever be 0 if the API starts at 1, so that probably means the version is unknown.

```suggestion
	if data.Version.IsUnknown() {
		var state RuleModel
		resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
		if resp.Diagnostics.HasErrors() {
			return
		}
		data.Version = state.Version
	}
```